executor/logger/worker_test.go

```diff
@@ -0,0 +1,218 @@
+C8OxVJ/At/IJGnWSIz8z+/YCUf7p4jd2LJgmZVVzXeDsOFcH62gu
+-----END RSA PRIVATE KEY-----`
+
+	caPEM = `-----BEGIN CERTIFICATE-----
+MIIDPDCCAiQCCQCBYUuEuypDMTANBgkqhkiG9w0BAQsFADBgMQswCQYDVQQGEwJV
+UzETMBEGA1UECAwKQ2FsaWZvcm5pYTEWMBQGA1UEBwwNU2FuIEZyYW5jaXNjbzEQ
+MA4GA1UECgwHU2VnbWVudDESMBAGA1UEAwwJbG9jYWxob3N0MB4XDTE3MTIyMzE1
+NTMxOVoXDTI3MTIyMTE1NTMxOVowYDELMAkGA1UEBhMCVVMxEzARBgNVBAgMCkNh
+bGlmb3JuaWExFjAUBgNVBAcMDVNhbiBGcmFuY2lzY28xEDAOBgNVBAoMB1NlZ21l
+bnQxEjAQBgNVBAMMCWxvY2FsaG9zdDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCC
+AQoCggEBAJwB+Yp6MyUepgtaRDxVjpMI2RmlAaV1qApMWu60LWGKJs4KWoIoLl6p
+oSEqnWrpMmb38pyGP99X1+t3uZjiK9L8nFhuKZ581tsTKLxaSl+YVg7JbH5LVCS6
+opsfB5ON1gJxf1HA9YyMqKHkBFh8/hdOGR0T6Bll9TPO1NQB/UqMy/tKr3sA3KZm
+XVDbRKSuUAQWz5J9/hLPmVMU41F/uD7mvyDY+x8GymInZjUXG4e0oq2RJgU6SYZ8
+mkscM6qhKY3mL487w/kHVFtFlMkOhvI7LIh3zVvWwgGSAoAv9yai9BDZNFSk0cEb
+bb/IK7BQW9sNI3lcnGirdbnjV94X9/sCAwEAATANBgkqhkiG9w0BAQsFAAOCAQEA
+MJLeGdYO3dpsPx2R39Bw0qa5cUh42huPf8n7rp4a4Ca5jJjcAlCYV8HzqOzpiKYy
+ZNuHy8LnNVYYh5Qoh8EO45bplMV1wnHfi6hW6DY5j3SQdcxkoVsW5R7rBF7a7SDg
+6uChVRPHgsnALUUc7Wvvd3sAs/NKHzHu86mgD3EefkdqWAaCapzcqT9mo9KXkWJM
+DhSJS+/iIaroc8umDnbPfhhgnlMf0/D4q0TjiLSSqyLzVifxnv9yHz56TrhHG/QP
+E/8+FEGCHYKM4JLr5smGlzv72Kfx9E1CkG6TgFNIHjipVv1AtYDvaNMdPF2533+F
+wE3YmpC3Q0g9r44nEbz4Bw==
+-----END CERTIFICATE-----`
+	keyPassword = ""
+)
+
+type certsConfig struct {
+	CertPEM     string
+	KeyPEM      string
+	KeyPassword string
+	CaPEM       string
+}
+
+func tlsConfig(t *testing.T) *certsConfig {
+	tmpDir := t.TempDir()
+
+	tmpFileCa := writeTmpFile(tmpDir, "test-ca.pem", []byte(caPEM))
+	tmpFileKey := writeTmpFile(tmpDir, "test-key.key", []byte(keyPEM))
+	tmpFileCert := writeTmpFile(tmpDir, "test-cert.cert", []byte(certPEM))
+
+	return &certsConfig{
+		CertPEM:     tmpFileCert.Name(),
+		KeyPEM:      tmpFileKey.Name(),
+		KeyPassword: keyPassword,
+		CaPEM:       tmpFileCa.Name(),
+	}
+}
+
+func writeTmpFile(dirname string, filename string, contents []byte) *os.File {
+	f, err := ioutil.TempFile(dirname, filename)
+	if err != nil {
+		log.Fatalf("cannot create temporary file %s/%s: %v", dirname, filename, err)
+	}
+
+	_, err = f.Write(contents)
+	if err != nil {
+		log.Fatalf("cannot write to temporary file %s/%s: %v", dirname, filename, err)
+	}
+
+	return f
+}
+
+func TestWorkerProducerLog(t *testing.T) {
+	config := tlsConfig(t)
+	p, err := kafka.NewProducer(&kafka.ConfigMap{
+		"bootstrap.servers":        "broker",
+		"go.delivery.reports":      false,
+		"debug":                    "all",
+		"socket.timeout.ms":        10,
+		"security.protocol":        "ssl",
+		"ssl.ca.location":          config.CaPEM,
+		"ssl.key.location":         config.KeyPEM,
+		"ssl.certificate.location": config.CertPEM,
+		"ssl.key.password":         config.KeyPassword,
+	})
+
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+	expectedLogs := map[struct {
+		tag     string
+		message string
+	}]bool{
+		{"INIT", "librdkafka"}: false,
+	}
+	go func() {
```

Comment:
:wrench: I suspect this test passes currently because it doesn't wait for the goroutine to complete - that just sits in the background somewhere.
When I run the for-loop directly within the test, it fails with a timeout panic:
```
|| panic: test timed out after 10s
|| .../go/pkg/mod/go.opencensus.io@v0.23.0/stats/view/worker.go:276 +0xb9
|| .../go/pkg/mod/go.opencensus.io@v0.23.0/stats/view/worker.go:34 +0x92
|| .../go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.6/pkg/log/log.go:63 +0x38
|| .../go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.6/pkg/log/log.go:62 +0x25
|| .../go/pkg/mod/github.com/confluentinc/confluent-kafka-go@v1.4.2/kafka/event.go:168 +0xcd
|| .../go/pkg/mod/github.com/confluentinc/confluent-kafka-go@v1.4.2/kafka/event.go:168 +0x106
|| .../go/pkg/mod/github.com/confluentinc/confluent-kafka-go@v1.4.2/kafka/producer.go:614 +0x52
|| .../go/pkg/mod/github.com/confluentinc/confluent-kafka-go@v1.4.2/kafka/producer.go:520 +0x29
|| .../go/pkg/mod/github.com/confluentinc/confluent-kafka-go@v1.4.2/kafka/producer.go:519 +0x692
|| .../go/pkg/mod/github.com/confluentinc/confluent-kafka-go@v1.4.2/kafka/producer.go:543 +0x45
|| .../go/pkg/mod/github.com/confluentinc/confluent-kafka-go@v1.4.2/kafka/producer.go:534 +0x26
|| .../go/pkg/mod/github.com/confluentinc/confluent-kafka-go@v1.4.2/kafka/producer.go:533 +0x70f
```

Although we could just tell Go to ignore this test, It's probably best to remove it as any proper integration tests will likely not care about the internal logs, but rather the ability to publish and then consume those published messages to check them.

Reply:
Indeed you're right, I removed it 😊